core/values/number.js

```diff
@@ -0,0 +1,83 @@
+const {
+    RuntimeError
+} = require('../error');
+const Value = require('./value');
+
+module.exports = class Number extends Value {
```

Comment:
I dont know that name of this class is good idea, because in js we have buildin class Number